c_emulator/riscv_sim.c

```diff
@@ -222,6 +228,7 @@ char *process_args(int argc, char **argv)
                     "V::"
                     "v::"
                     "l:"
+                    "c:"
```

Comment:
```suggestion
#ifdef SAILCOV
                    "c:"
#endif
```

Reply:
Yes, I wondered about this, but "r" is similar in being guarded later by an ifdef but not guarded here. Should that also be fixed?

Reply:
Yes, otherwise it (a) tries to parse the option's argument and gives an error if not present (b) ignores the option once parsed rather than giving an error:
```
burn:~ jrtc4% /local/scratch/jrtc4/cheri/sail-cheri-riscv/c_emulator/cheri_riscv_sim_RV64 -r
/local/scratch/jrtc4/cheri/sail-cheri-riscv/c_emulator/cheri_riscv_sim_RV64: option requires an argument -- 'r'
Usage: /local/scratch/jrtc4/cheri/sail-cheri-riscv/c_emulator/cheri_riscv_sim_RV64 [options] <elf_file>
	 -d	 --enable-dirty-update
	 -m	 --enable-misaligned
	 -P	 --enable-pmp
	 -z	 --ram-size
	 -C	 --disable-compressed
	 -I	 --disable-writable-misa
	 -F	 --disable-fdext
	 -i	 --mtval-has-illegal-inst-bits
	 -b	 --device-tree-blob
	 -t	 --terminal-log
	 -p	 --show-times
	 -a	 --report-arch
	 -T	 --test-signature
	 -h	 --help
	 -v	 --trace
	 -V	 --no-trace
	 -l	 --inst-limit
burn:~ jrtc4% /local/scratch/jrtc4/cheri/sail-cheri-riscv/c_emulator/cheri_riscv_sim_RV64 -r 0
No elf file provided.
Usage: /local/scratch/jrtc4/cheri/sail-cheri-riscv/c_emulator/cheri_riscv_sim_RV64 [options] <elf_file>
	 -d	 --enable-dirty-update
	 -m	 --enable-misaligned
	 -P	 --enable-pmp
	 -z	 --ram-size
	 -C	 --disable-compressed
	 -I	 --disable-writable-misa
	 -F	 --disable-fdext
	 -i	 --mtval-has-illegal-inst-bits
	 -b	 --device-tree-blob
	 -t	 --terminal-log
	 -p	 --show-times
	 -a	 --report-arch
	 -T	 --test-signature
	 -h	 --help
	 -v	 --trace
	 -V	 --no-trace
	 -l	 --inst-limit
```

Reply:
Ouch, I'll add the fixes for this and RVFI to this PR :)

Reply:
Done, hope this commit structure makes sense.